### ISSUE TITLE
YAML config and parser

### ISSUE THREAD

**Issue by mooster531** [bh2smith/dune-sync#38]
The TOML format, although simpler and with fewer pitfalls than YAML, simply isn't flexible enough to cover our use cases while still remaining readable.

Switch the configuration to YAML format, then invent a proper, well-structured config file format that covers at least the use cases we've defined and are aware of so far.

**Comment by mooster531** [comment]
Proposal for new format of config:
```yaml
jobs:
  - name: Download simple test query to local postgres
    source:
      ref: dune
      query_id: "4238114"
      query_engine: "medium"
      poll_frequency: 5
      parameters:
        - name: "blockchain"
          value: "gnosis"
          type: "ENUM"
        - name: "block_time"
          value: "2024-09-01 00:00:00"
          type: "DATE"
        - name: "result_limit"
          value: "10"
          type: "NUMBER"
    destination:
      ref: my_postgres
      table_name: "parameterized_results_4238114"
      if_exists: "replace"

  - name: Some other job
    source:
      ref: postgres
      table_name: foo_table
      if_exists: "append"
    destination:
      ref: dune
      table_name: "dune.username.table_name"
```

**Comment by bh2smith** [comment]
All looks great here - although I don't know what "my_postgres" is meant to represent.

In fact I am beginning to see why we might want to keep DB URL together with the config... since there may be multiple postgres instances. 
However we should not worry about this right now.

**Comment by mooster531** [comment]
The second paragraph of your comment is the answer to the question you posed in the first :) 

I'm thinking `my_postgres` could be a reference to an as-of-yet nonexistent key in that config which would make the config look like this:
```yaml

jobs:
    [...]
backends:
    - name: my_postgres
      type: postgres
      dsn: "postgresql://user:password@..."
      
    - name: other_postgres
    ...
    
```

and so on.

**Comment by bh2smith** [comment]
Closed by #40
